File: fsharp/bob/Bob.fs
module Bob
open System

let response (input: string): string = 
    let isShouting (input: string): bool = Seq.exists Char.IsLetter input && input.ToUpper() = input
    let isQuestion (input: string): bool = input.EndsWith("?")
    let isForcefulQuestion (input: string): bool = isShouting(input) && isQuestion(input)
    let trimmed = input.Trim()

    match input with
        | _ when trimmed |> String.IsNullOrWhiteSpace -> "Fine. Be that way!"
        | _ when trimmed |> isForcefulQuestion -> "Calm down, I know what I'm doing!"
        | _ when trimmed |> isShouting -> "Whoa, chill out!"
        | _ when trimmed |> isQuestion -> "Sure."
        | _ -> "Whatever."

File: csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string input)
    {
        if (string.IsNullOrWhiteSpace(input))
        {
            return "Fine. Be that way!";
        }
        if (IsShouting(input))
        {
            return "Whoa, chill out!";
        }
        if (IsQuestion(input))
        {
            return "Sure.";
        }
        return "Whatever.";
    }

    private static bool IsQuestion(string input)
    {
        return input.Trim().EndsWith("?");
    }

    private static bool IsShouting(string input)
    {
        return input.Any(char.IsLetter) && input.Where(char.IsLetter).All(char.IsUpper);
    }
}